subsys/bluetooth/audio/has.c

```diff
@@ -194,9 +202,40 @@ static void security_changed(struct bt_conn *conn, bt_security_t level, enum bt_
 	client = client_get_or_new(conn);
 	if (unlikely(!client)) {
 		BT_ERR("Failed to allocate client");
+		return;
+	}
+
+	/* Notify after reconnection */
+	if (client->ntf_bonded.in_progress) {
+		/* Emit active preset notification */
+		k_work_submit(&active_preset_work);
+
+		/* Emit preset changed notifications */
+		k_work_submit(&client->control_point_work);
 	}
 }
 
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	struct has_client *client;
+
+	BT_DBG("conn %p err %d", conn, err);
+
+	if (err != 0 || !bt_addr_le_is_bonded(conn->id, &conn->le.dst)) {
+		return;
+	}
+
+	client = client_get_or_new(conn);
+	if (unlikely(!client)) {
+		BT_ERR("Failed to allocate client");
+		return;
+	}
+
+	/* Mark preset changed operation as pending */
+	client->ntf_bonded.in_progress = true;
```

Comment:
Should this be unconditional, or only if the client has actually subscribed to the characteristic?

Reply:
Yeah, I could check the subscription state here... But the notification would not be sent anyway, because the check is performed in another function prior sending PDU